src/js/components/table.js

```diff
@@ -0,0 +1,241 @@
+const select = require("../utils/select");
+const behavior = require("../utils/behavior");
+const { CLICK } = require("../events");
+const { prefix: PREFIX } = require("../config");
+
+const TABLE = `.${PREFIX}-table`;
+const SORTED = "aria-sort";
+const ASCENDING = "ascending";
+const DESCENDING = "descending";
+const SORT_OVERRIDE = "data-sort-value";
+
+const ICON_SOURCE = `
+  <svg class="${PREFIX}-icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+    <g class="descending" fill="transparent">
+      <path d="M19,15l-1.41-1.41L13,18.17V2H11v16.17l-4.59-4.59L5,15l7,7L19,15z"/>
+    </g>
+    <g class="ascending" fill="transparent">
+      <path transform="rotate(180, 12, 12)" d="M19,15l-1.41-1.41L13,18.17V2H11v16.17l-4.59-4.59L5,15l7,7L19,15z"/>
+    </g>
+    <g class="unsorted" fill="transparent">
+      <polygon points="15.17 15 13 17.17 13 6.83 15.17 9 16.58 7.59 12 3 7.41 7.59 8.83 9 11 6.83 11 17.17 8.83 15 7.42 16.41 12 21 16.59 16.41 15.17 15"/>
+    </g>
+  </svg>
+`;
+
+const SORT_BUTTON_CLASS = `${PREFIX}-table__header__button`;
+const SORT_BUTTON = `.${SORT_BUTTON_CLASS}`;
+const SORTABLE_HEADER = `th[data-sortable]`;
+const ANNOUNCEMENT_REGION = `.${PREFIX}-table__announcement-region[aria-live="polite"]`;
+
+
+/** Gets the data-sort-value attribute value, if provided — otherwise, gets
+ * the innerText or textContent — of the child element (HTMLTableCellElement)
+ * at the specified index of the given table row
+ *
+ * @param {number} index
+ * @param {array<HTMLTableRowElement>} tr
+ * @return {boolean}
+ */
+const getCellValue = (tr, index) => tr.children[index].getAttribute(SORT_OVERRIDE)
+  || tr.children[index].innerText
+  || tr.children[index].textContent;
+
+/**
+ * Compares the values of two row array items at the given index, then sorts by the given direction
+ * @param {number} index
+ * @param {string} direction
+ * @return {boolean}
+ */
+const compareFunction = (index, isAscending) => (thisRow, nextRow) => {
+
+  // get values to compare from data attribute or cell content
+  let value1 = getCellValue(isAscending ? thisRow : nextRow, index);
+  let value2 = getCellValue(isAscending ? nextRow : thisRow, index);
+
+  // if neither value is empty, and if both values are already numbers, compare numerically
+  if (value1 !== '' && value2 !== '' && !Number.isNaN(Number(value1)) && !Number.isNaN(Number(value2))) {
```

Comment:
If we're checking if it's not empty couldn't we do something like:
```js
  if (value1 && value2 && !Number.isNaN(Number(value1)) && !Number.isNaN(Number(value2)))
```

And get the same result?

Reply:
Your suggestions, as always, make the code better :) Thanks James.